zipkin-query-service/src/main/scala/com/twitter/zipkin/query/Main.scala

```diff
@@ -35,11 +36,15 @@ object Main {
     val source = if (runtime.configFile.exists()) Files.toString(runtime.configFile, UTF_8)
     else Resources.toString(getClass.getResource(runtime.configFile.toString), UTF_8)
 
-    val builder = (new Eval).apply[Builder[RuntimeEnvironment => ZipkinQuery]](source)
+    val builder = (new Eval).apply[Builder[RuntimeEnvironment => ListeningServer]](source)
     try {
-      val server = builder.apply().apply(runtime)
-      server.start()
-      ServiceTracker.register(server)
+      val query = builder.apply().apply(runtime)
+      Await.ready(query)
+      ServiceTracker.register(new Service() {
+        override def shutdown() = Await.ready(query.close())
+
+        override def start() = {}
+      })
```

Comment:
Yeah, it is shoe-horning. Migrating zipkin-collector-server and zipkin-query-server off ostrich is yet another todo :) Where to, is yet another question! finatra, finch etc.